apps/emqx_utils/src/emqx_placeholder.erl

```diff
@@ -257,28 +257,8 @@ quote_mysql(Str) ->
 
 lookup_var(Var, Value) when Var == ?PH_VAR_THIS orelse Var == [] ->
     Value;
-lookup_var([Prop | Rest], Data) ->
-    case lookup(Prop, Data) of
-        {ok, Value} ->
-            lookup_var(Rest, Value);
-        {error, _} ->
-            undefined
-    end.
-
-lookup(Prop, Data) when is_binary(Prop) ->
-    case maps:get(Prop, Data, undefined) of
-        undefined ->
-            try
-                {ok, maps:get(binary_to_existing_atom(Prop, utf8), Data)}
-            catch
-                error:{badkey, _} ->
-                    {error, undefined};
-                error:badarg ->
-                    {error, undefined}
-            end;
-        Value ->
-            {ok, Value}
-    end.
+lookup_var(Var, Data) ->
+    emqx_rule_maps:nested_get(Var, Data).
```

Comment:
Now `emqx_utils` depend on `emqx_rule_engine`. I specifically tried to avoid it before, thus this simple implementation.

I'd propose to just introduce another clause here, like in `emqx_rule_maps`.
https://github.com/emqx/emqx/blob/a757afcb7eca30c41fa049f70da4d2e4fc636612/apps/emqx_rule_engine/src/emqx_rule_maps.erl#L88-L89

<sup>(though I think that will muddle the interface slightly, and will have to hide some error conditions, it's still probably much simpler than having to introduce JSON decoding in all of this module's clients)</sup>